packages/cli/src/commonlib/telemetry.ts

```diff
@@ -31,22 +32,40 @@ export class CliTelemetryReporter implements TelemetryReporter {
     if (rootPath) {
       this.rootFolder = rootPath;
       this.reporter.setAppRoot(rootPath);
+
+      // add shared properties
+      const fixedProjectSettings = getFixedCommonProjectSettings(rootPath);
+      this.addSharedProperty(TelemetryProperty.ProjectId, fixedProjectSettings?.projectId ?? "");
```

Comment:
instead of add lots of check here. How about move the check string part into addSharedProperty method?

Reply:
Fixed, thx!